### DISCUSSION TITLE
Fast-forward and fast-rewind advanced experience

### DISCUSSION THREAD

**Discussion by matanox** [videojs/video.js#7472] [Q&A] [answered]
Hi, 

I would like to learn whether this player supports going back and forward in video while providing a fast forward and fast-backward play experience, i.e. the experience whereby the video really renders very fast from the current video position all the way to the desired new position while fast-playing the video between the two points. (obviously doing so by constantly skipping many frames at the desired rate). 

Frankly I am not aware of any player that actually does that, other than what you would occasionally see in TV series and in movies. Also, perhaps you can point me at library API for this project for jumping around the video programmatically rather than through a user interface control. 

Many thanks,
Matan

**Comment by gkatsev** [comment, marked as answer]
You definitely can. For HTML5 playback, there's a playback rate you can set. This is available via `player.playbackRate()`, it can take a number between 0 and 16, in most browsers. In safari, you can also give it a negative number for it to play backwards.
Hope this helps.

**Reply by matanox** [reply]
Thanks, I guess you mean this. https://docs.videojs.com/player#playbackRate. This might work akin to fast forward, although I guess there is no _backward_ playback.

**Reply by gkatsev** [reply]
Yup, that it. While the spec does allow for backwards playback, it's only available in Safari and it looks like Firefox and Chrome are unlikely to implement it in the near term.

**Comment by matanox** [comment]
Yes, thanks a lot! does it also happen to support going back and forward frame by frame? What would it take for laymen to implement a plugin for that? Can a plugin control playback rate as well as going back and forward beyond the specific experience already provided via the [playbackrates option](https://github.com/videojs/video.js/blob/main/docs/guides/options.md#playbackrates)? Apologies for the edits.
